src/ui/views/ConfiguratorView/index.tsx

```diff
@@ -147,10 +144,20 @@ enum ViewState {
 
 interface ConfiguratorViewProps {
   gitRepository: GitRepository;
+  selectedDevice: string | null;
+  multicastDnsDevices: {
```

Comment:
It might be better replace `multicastDnsDevices` indexed object type annotation with `Record<string, MulticastDnsInformation>` or `Map<string, MulticastDnsInformation>`.

Reply:
You could also define your own type for this since you are using this data structure at least in 2 places

Reply:
I have converted it to Map<string, MulticastDnsInformation>